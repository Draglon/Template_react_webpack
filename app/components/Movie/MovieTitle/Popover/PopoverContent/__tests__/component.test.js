import React from 'react';

import PopoverContentComponent from '../component';

describe('<PopoverContentComponent />', () => {
  const props = {
    closePopover: () => {},
    myLists: {
      results: [],
    },
    movieId: 999,
  };
  const component = shallow(<PopoverContentComponent {...props} />);

  describe('without results', () => {
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
  describe('with results', () => {
    it('should match snapshot', () => {
      const newProps = {
        ...props,
        myLists: {
          results: [{ id: 1, name: 'List 1' }, { id: 1, name: 'List 1' }],
        },
      };
      component.setProps(newProps);
      expect(component).toMatchSnapshot();
    });
  });
});
