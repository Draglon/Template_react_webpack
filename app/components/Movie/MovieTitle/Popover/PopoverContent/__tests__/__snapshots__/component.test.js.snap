// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopoverContentComponent /> with results should match snapshot 1`] = `
<nav
  className="popover__nav"
>
  <ul>
    <li
      onClick={[Function]}
    >
      <Connect(CreateListModalContainer)
        text="Create new list ..."
      />
    </li>
    <li
      key="1"
    >
      <Connect(PopoverNavLinkContainer)
        closePopover={[Function]}
        list={
          Object {
            "id": 1,
            "name": "List 1",
          }
        }
        movieId={999}
      />
    </li>
    <li
      key="1"
    >
      <Connect(PopoverNavLinkContainer)
        closePopover={[Function]}
        list={
          Object {
            "id": 1,
            "name": "List 1",
          }
        }
        movieId={999}
      />
    </li>
  </ul>
</nav>
`;

exports[`<PopoverContentComponent /> without results should match snapshot 1`] = `
<nav
  className="popover__nav"
>
  <ul>
    <li
      onClick={[Function]}
    >
      <Connect(CreateListModalContainer)
        text="Create new list ..."
      />
    </li>
  </ul>
</nav>
`;
